dspy/clients/lm.py

```diff
@@ -163,54 +225,135 @@ def copy(self, **kwargs):
         return new_instance
 
 
-@functools.lru_cache(maxsize=None)
-def cached_litellm_completion(request, num_retries: int):
-    return litellm_completion(
-        request,
+class LMRequestLRUCache(OrderedDict):
```

Comment:
is there an off-the-shelf async lru cache which can be used? https://github.com/aio-libs/async-lru for example

Reply:
I thought of that first but because Omar wanted the async and sync to share the same cache I implemented it this way